cmsplugin_filer_image/cms_plugins.py

```diff
@@ -147,106 +171,94 @@ class Media:
-                height = instance.thumbnail_option.height
-            crop = instance.thumbnail_option.crop
-            upscale = instance.thumbnail_option.upscale
-        else:
-            if placeholder_width:
-                # use the placeholder width as a hint for sizing
-                width = int(placeholder_width)
-            if placeholder_height:
-                height = int(placeholder_height)
-
-        if instance.has_attached_image():
-            if instance.image.subject_location:
-                subject_location = instance.image.subject_location
-            if not height and width:
-                # height was not externally defined: use ratio to scale it by the width
-                height = int( float(width)*float(instance.image.height)/float(instance.image.width) )
-            if not width and height:
-                # width was not externally defined: use ratio to scale it by the height
-                width = int( float(height)*float(instance.image.width)/float(instance.image.height) )
-            if not width:
-                # width is still not defined. fallback the actual image width
-                width = instance.image.width
-            if not height:
-                # height is still not defined. fallback the actual image height
-                height = instance.image.height
-        return {'size': (width, height),
-                'crop': crop,
-                'upscale': upscale,
-                'subject_location': subject_location}
+        }
 
     def get_thumbnail(self, context, instance):
         if instance.has_attached_image():
             filer_file_field = instance.image.image.file
-            thumbnail_options = self._get_thumbnail_options(context, instance)
+            thumbnail_options = get_thumbnail_options(context, instance)
             return filer_file_field.get_thumbnail(thumbnail_options)
 
-    def _get_default_horiz_space(self, instance, context):
-        if ("inherited_from_parent" in context and
-            not instance.horiz_space and
-            not instance.alignment == instance.CENTER):
-            return instance.DEFAULT_HORIZONTAL_SPACE
+    def render(self, context, instance, placeholder):
+        """
+        Render the context needed by the plugin template.
+        Preffer to evaluate all attributes here so hidden/lazy operations can be profiled.
+        """
+        if not instance.image:
+            return {}
+        options = get_thumbnail_options(context, instance)
+        img_size = options["size"]
+
+        if instance.image.width == img_size[0] and instance.image.height == img_size[1]:
+            image = instance.image
         else:
-            return ''
+            image = get_thumbnailer(instance.image).get_thumbnail(options)
 
-    def render(self, context, instance, placeholder):
-        options = self._get_thumbnail_options(context, instance)
+        container_classes = ""
+        container_attributes = ""
```

Comment:
No, we addedd data-rel=<img-link> for the overlay feature but that is no longer required. Thanks.